### PR TITLE
provide `@permutation_group` and `permutation_group`

### LINKED ISSUE #1907: Add a more convenient way to enter permutation groups
In GAP I can write
```gap
H := Group( (1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8) );
```

In Magma this is:
```
H := PermutationGroup< 9 | (1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8) >;
```

In OSCAR the best I could come up is this:
```julia
H = sub(symmetric_group(9), @perm 9 [(1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8)])[1]
```

I'd like to improve this, in two ways:

1. add a function which directly takes a vector of permutations and constructs a permutation groups, with a given degree:
   ```julia
   H = permutation_group(9, @perm [(1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8)])
   ```
2. add a macro variant of this:
   ```julia
   H = @permutation_group(9, (1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8))
   # optionally could use different syntax, e.g.:
   H = @permutation_group(9 ; (1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8))
   H = @permutation_group(9 | (1,2,4)(5,6,8)(3,9,7), (4,5,6)(7,9,8))
   ```

Thoughts, comments, suggestions?

### PR DESCRIPTION
**Pull request by ThomasBreuer**
resolves #1907

### CODE REVIEW

**Review by fingolfin — APPROVED** [review]
Great, thank you!

**Comment by fingolfin** [comment]
I'll wait and see if @joschmitt and @thofma are happy with this (of course we can still improve it further in future PRs anyway)

File: src/Groups/perm.jl [hunk 1]
```diff
@@ -661,3 +664,51 @@ macro perm(n,gens)
        end
     end
 end
+
+
+@doc raw"""
+    permutation_group(n::T, perms::Vector{PermGroupElem}) where T <: IntegerUnion
```
Comment (thofma) [inline src/Groups/perm.jl]:
```suggestion
    permutation_group(n::IntegerUnion, perms::Vector{PermGroupElem})
```

File: src/Groups/perm.jl [hunk 2]
```diff
@@ -661,3 +664,51 @@ macro perm(n,gens)
...
+
+# Examples
+```jldoctest
+julia> x = cperm([1,2,3], [4,5]);  y = cperm([1,4]);
+
+julia> Oscar.permutation_group(5, [x, y])
+Group([ (1,2,3)(4,5), (1,4) ])
+```
+"""
+function permutation_group(n::T, perms::Vector{PermGroupElem}) where T <: IntegerUnion
```
Comment (thofma) [inline src/Groups/perm.jl]:
```suggestion
function permutation_group(n::IntegerUnion, perms::Vector{PermGroupElem})
```

**Comment by thofma** [comment]
Fine with me. I left two minor comments.

File: src/Groups/perm.jl [hunk 3]
```diff
@@ -661,3 +664,51 @@ macro perm(n,gens)
...
+    permutation_group(n::T, perms::Vector{PermGroupElem}) where T <: IntegerUnion
+
+Return the permutation group of degree `n` that is generated by the
+elements in `perms`.
+
+# Examples
+```jldoctest
+julia> x = cperm([1,2,3], [4,5]);  y = cperm([1,4]);
+
+julia> Oscar.permutation_group(5, [x, y])
```
Comment (joschmitt) [inline src/Groups/perm.jl]:
```suggestion
julia> permutation_group(5, [x, y])
```

**Review by joschmitt — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**